build/tools/SharedCodeGenerator/Program.cs

```diff
@@ -106,30 +107,100 @@ private static void GenerateSupportedPlatformsReadmeFile(string repoDir)
+                                sw.WriteLine();
+                                AddVersions(sw, Path.Join(platformSubDirPath, "versions"));
+                                break;
+                        }
+
+                        sw.WriteLine();
+                    }
+                }
+
+                sw.Flush();
+            }
+        }
+
+        /// <summary>
+        /// Writes all versions found in text files in a directory
+        /// to a stream, in order of version.
+        /// </summary>
+        /// <param name="sw">stream writer instance to write the versions to</param>
+        /// <param name="versionsPath">directory at which the versionsToBuild.txt and optional legacyVersions.txt file is found.</param>
+        private static void AddVersions(StreamWriter sw, string versionsPath)
+        {
+            foreach (var osTypeDirPath in Directory.GetDirectories(versionsPath))
+            {
+                var osTypeDirInfo = new DirectoryInfo(osTypeDirPath);
+                var osType = osTypeDirInfo.Name;
+                sw.WriteLine($"### {osType}");
+                sw.WriteLine();
+                var versionFile = Path.Join(osTypeDirPath, "versionsToBuild.txt");
+                var legacyVersionFile = Path.Join(osTypeDirPath, "legacyVersions.txt");
+                var supportedVersions = new List<string>();
+                using (var reader = new StreamReader(versionFile))
+                {
+                    string line;
+                    while ((line = reader.ReadLine()) != null)
+                    {
+                        if (string.IsNullOrWhiteSpace(line) || line.Trim().StartsWith("#"))
+                        {
+                            continue;
+                        }
+
+                        var parts = line.Split(",", StringSplitOptions.RemoveEmptyEntries);
+                        var versionPart = parts[0];
+                        supportedVersions.Add(versionPart);
+                    }
+                }
+
+                if (File.Exists(legacyVersionFile))
+                {
+                    using (var reader = new StreamReader(legacyVersionFile))
                     {
                         string line;
                         while ((line = reader.ReadLine()) != null)
                         {
-                            if (string.IsNullOrWhiteSpace(line) || line.Trim().StartsWith("#"))
+                            line = line.Trim();
+
+                            if (string.IsNullOrWhiteSpace(line) || line.StartsWith("#"))
                             {
                                 continue;
                             }
 
-                            var parts = line.Split(",", StringSplitOptions.RemoveEmptyEntries);
-                            var versionPart = parts[0];
-                            sw.WriteLine($"- {versionPart}");
+                            supportedVersions.Add(line);
```

Comment:
Do we need to modify the `line` in any way before adding it to `supportedVersions`? Are the lines written to `legacyVersions.txt` guaranteed to always be _just_ a version from what you've seen?